core/src/main/scala/caliban/execution/QueryExecution.scala

```diff
@@ -3,29 +3,39 @@ package caliban.execution
 /**
  * Defines which type of parallelism to use when executing queries
  */
-sealed trait QueryExecution
+sealed trait QueryExecution {
+  def tag: Int
```

Comment:
This is the pattern that was removed in ZIO and ZPure. Is it beneficial here?

Reply:
Yes, but unlike in ZIO / ZPure here we call the `tag` only once per execution (instead of once per iteration). This way the penalty is negligible